src/jit/codegencommon.cpp

```diff
@@ -10317,10 +10333,23 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     compiler->unwindBegProlog();
 
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
+
     // TODO Save callee-saved registers
 
     // This is the end of the OS-reported prolog for purposes of unwinding
     compiler->unwindEndProlog();
+
+    // Restore frame pointer
+    // TODO We will need changes here when we introduce PSPSym
+    inst_RV_RV(INS_mov, REG_FPBASE, REG_ARG_0);
```

Comment:
Sorry for the confusion. What I meant was there is no CallEHFunclet for the second throw. For the first throw, the backtrace looks OK.

```
(gdb) x/20i CallEHFunclet                                                                                                             [63/9485]
   0xf719af9c <CallEHFunclet>:  push   %ebp
   0xf719af9d <CallEHFunclet+1>:        mov    %esp,%ebp
   0xf719af9f <CallEHFunclet+3>:        push   %ebx
   0xf719afa0 <CallEHFunclet+4>:        push   %esi
   0xf719afa1 <CallEHFunclet+5>:        push   %edi
   0xf719afa2 <CallEHFunclet+6>:        mov    0x14(%ebp),%eax
   0xf719afa5 <CallEHFunclet+9>:        mov    %esp,(%eax)
   0xf719afa7 <CallEHFunclet+11>:       mov    0xc(%ebp),%edx
   0xf719afaa <CallEHFunclet+14>:       mov    0x8(%ebp),%eax
   0xf719afad <CallEHFunclet+17>:       mov    0x10(%ebp),%ecx
   0xf719afb0 <CallEHFunclet+20>:       mov    (%ecx),%edi
   0xf719afb2 <CallEHFunclet+22>:       mov    0x4(%ecx),%esi
   0xf719afb5 <CallEHFunclet+25>:       mov    0x8(%ecx),%ebx
   0xf719afb8 <CallEHFunclet+28>:       mov    0x18(%ecx),%ebp
=> 0xf719afbb <CallEHFunclet+31>:       call   *%edx
   0xf719afbd <CallEHFunclet+33>:       pop    %edi
   0xf719afbe <CallEHFunclet+34>:       pop    %esi
   0xf719afbf <CallEHFunclet+35>:       pop    %ebx
   0xf719afc0 <CallEHFunclet+36>:       pop    %ebp
   0xf719afc1 <CallEHFunclet+37>:       ret    $0x10
(gdb) bt
#0  CallEHFunclet () at ehhelpers.S:157
#1  0xf7176612 in ExceptionTracker::CallHandler (this=0x8081d40, uHandlerStartPC=4066256066, sf=..., pEHClause=0x8081d94, pMD=0xf6335988, funcletType=Catch, pContextRecord=0xffffc398) at /home/hanjoung/ws/dotnet/coreclr/src/vm/exceptionhandling.cpp:3314
#2  0xf7176095 in ExceptionTracker::CallCatchHandler (this=0x8081d40, pContextRecord=0xffffc398, pfAborting=0xffffbd17) at /home/hanjoung/ws/dotnet/coreclr/src/vm/exceptionhandling.cpp:583
#3  0xf7178833 in ProcessCLRException (pExceptionRecord=0x80e7c20, MemoryStackFp=4294954048, pContextRecord=0xffffc398, pDispatcherContext=0xffffbee8) at /home/hanjoung/ws/dotnet/coreclr/src/vm/exceptionhandling.cpp:1104
#4  0xf71832b8 in UnwindManagedExceptionPass2 (ex=..., unwindStartContext=0xffffc398) at /home/hanjoung/ws/dotnet/coreclr/src/vm/exceptionhandling.cpp:4394
#5  0xf7183c6e in UnwindManagedExceptionPass1 (ex=..., frameContext=0xffffc728) at /home/hanjoung/ws/dotnet/coreclr/src/vm/exceptionhandling.cpp:4548
#6  0xf718472e in DispatchManagedException (ex=..., isHardwareException=false) at /home/hanjoung/ws/dotnet/coreclr/src/vm/exceptionhandling.cpp:4655
#7  0xf700060e in IL_Throw (obj=0xf30149bc) at /home/hanjoung/ws/dotnet/coreclr/src/vm/jithelpers.cpp:5297
#8  0xf25e24b7 in ?? ()
#9  0xf719b113 in CallDescrWorkerInternal () at asmhelpers.S:442
#10 0xf6f17f18 in CallDescrWorker (pCallDescrData=0xffffd2a8) at /home/hanjoung/ws/dotnet/coreclr/src/vm/callhelpers.cpp:135
#11 0xf6f17cbc in CallDescrWorkerWithHandler (pCallDescrData=0xffffd2a8, fCriticalCall=0) at /home/hanjoung/ws/dotnet/coreclr/src/vm/callhelpers.cpp:78
#12 0xf6f19d33 in MethodDescCallSite::CallTargetWorker (this=0xffffd428, pArguments=0xffffd3d0, pReturnValue=0xffffd320, cbReturnValue=8) at /home/hanjoung/ws/dotnet/coreclr/src/vm/callhelpers.cpp:645
#13 0xf6e9c097 in MethodDescCallSite::Call_RetArgSlot (this=0xffffd428, pArguments=0xffffd3d0) at /home/hanjoung/ws/dotnet/coreclr/src/vm/callhelpers.h:433
#14 0xf71f8b42 in RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::$_1::operator()(RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::Param*) const::{lambda(RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::Param*)#1}::operator()(RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::Param*) const (this=0xffffd4f0, pParam=0xffffd5a8) at /home/hanjoung/ws/dotnet/coreclr/src/vm/assembly.cpp:2061
#15 0xf71f5110 in RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::$_1::operator()(RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::Param*) const (this=0xffffd5a0, __EXparam=0xffffd5a8) at /home/hanjoung/ws/dotnet/coreclr/src/vm/assembly.cpp:2075
#16 0xf71f4eed in RunMain (pFD=0xf6335988, numSkipArgs=1, piRetVal=0xffffd664, stringArgs=0xffffd9f8) at /home/hanjoung/ws/dotnet/coreclr/src/vm/assembly.cpp:2075
#17 0xf71f5420 in Assembly::ExecuteMainMethod (this=0x80b7100, stringArgs=0xffffd9f8, waitForOtherThreads=1) at /home/hanjoung/ws/dotnet/coreclr/src/vm/assembly.cpp:2157
#18 0xf6cc6c91 in CorHost2::ExecuteAssembly (this=0x805a190, dwAppDomainId=1, pwzAssemblyPath=0x80a11f0 u"/work/nfs/dotnet/misctest/eh1/eh.dll", argc=0, argv=0x0, pReturnValue=0xffffdbfc) at /home/hanjoung/ws/dotnet/coreclr/src/vm/corhost.cpp:501
#19 0xf6c51094 in coreclr_execute_assembly (hostHandle=0x805a190, domainId=1, argc=0, argv=0x0, managedAssemblyPath=0x805209c "/work/nfs/dotnet/misctest/eh1/eh.dll", exitCode=0xffffdbfc) at /home/hanjoung/ws/dotnet/coreclr/src/dlls/mscoree/unixinterface.cpp:376
#20 0x0804c624 in ExecuteManagedAssembly (currentExeAbsolutePath=0x8052014 "/work/nfs/dotnet/tt/corerun", clrFilesAbsolutePath=0x8052074 "/work/nfs/dotnet/tt", managedAssemblyAbsolutePath=0x805209c "/work/nfs/dotnet/misctest/eh1/eh.dll", managedAssemblyArgc=0, managedAssemblyArgv=0x0) at /home/hanjoung/ws/dotnet/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:420
#21 0x0804b1e1 in corerun (argc=2, argv=0xffffddf4) at /home/hanjoung/ws/dotnet/coreclr/src/coreclr/hosts/unixcorerun/corerun.cpp:149
#22 0x0804b2ea in main (argc=2, argv=0xffffddf4) at /home/hanjoung/ws/dotnet/coreclr/src/coreclr/hosts/unixcorerun/corerun.cpp:161
```